README.md

```diff
@@ -264,13 +269,92 @@ Command Line Flag | Environment Variable | Default | Description
 
 ## Example App
 
-There is an example Servant application implementing all of the authentiation flows described above.  You can find it in the [example-servant-app](./example-servant-app/) directory.
+There is an example Servant application implementing all of the authentication
+flows described above.  You can find it in the
+[example-servant-app](./example-servant-app/) directory.
+
+## Future Work
+
+There are many features we would like to add to Goat Guardian at some point.
+The following are just a small taste.  You can find many more features on the
+[issue tracker](https://github.com/arow-oss/goat-guardian/issues).
+
+- Support other OAuth providers like Google, Github, and Facebook (#1)
+
+- Allow a user to login with *either* email or OAuth and link the accounts.
+
+- Provide a way for Goat Guardian to store user data in a database other than SQLite (#14)
+
+- Allow Goat Guardian settings to be specified in a configuration file (#17)
+
+- Specify "routing-rules" for Goat Guardian, in order to serve static files as
+  well as a web application.  This could also be used to allow Goat Guardian to
+  reverse proxy for multiple web applications, so that users could easy login to
+  multiple related microservices (#18)
+
+- Make Goat Guardian completely programmable, by providing it as a Haskell
+  library (instead of only as a standalone executable).  Eventually we also
+  want to provide access to it as a C library, so that it can be programmed
+  from any programming language, including Python, Ruby, Java, etc.
```

Comment:
This feature would be geared towards advanced users.

I am imagining a scenario where a team starts using Goat Guardian to do authentication for their web application.  After a while, they end up wanting to make some small changes to Goat Guardian.  At first, they are able to specify command line arguments or environment variables to make the changes they want.  

However, at some point they want to make changes that we haven't thought of.  For instance, maybe they want to set an additional HTTP Header after the user logs in.

If we make Goat Guardian programmable as a Haskell library (similar to XMonad), then the team could make a small change using the API provided.  They could continue using Goat Guardian without having to spend much effort.

However, if Goat Guardian is just a standalone executable, then the team would either have to fork the Goat Guardian source code to make their changes, or start handling authentication in their own application.

---------------------------------------------------------------------------

That being said, I think it would require quite a lot of work to enable Goat Guardian to be used as a Haskell library.  Maybe we should hope that a user on Github sends us a PR for it.

Reply:
I'm not sure, but I think it's better to provide auth libraries directly for each language (e.g., `servan-auth`) rather than using our limited life time to create each language version of Goat Guardian library.

Also, one of the best benefit of Goat Guardian is that it is independent from programming languages that the developer can use, so at least I guess it is not the advantage if Goat Guardian require developers to program in specific language.

Of course it would be OK to provide *only* Haskell library for advanced developers, but there would be no reasons to provide other language versions.

Reply:
Okay, that sounds good.  I'll go ahead and remove this point.